pkg/collector/corechecks/cluster/topologycollectors/container_correlator.go

```diff
@@ -91,12 +93,14 @@ func (cc *ContainerCorrelator) CorrelateFunction() error {
 				containerPort = cntPort
 			}
 
-			if nodeIdentifier, ok := nodeMap[pod.NodeName]; ok {
+			if nodeCorrelation, ok := nodeMap[pod.NodeName]; ok {
 				// submit the StackState component for publishing to StackState
-				containerComponent := cc.containerToStackStateComponent(nodeIdentifier, pod, container, containerPort)
+				containerComponent := cc.containerToStackStateComponent(nodeCorrelation.NodeIdentifier, pod, container, containerPort)
 				cc.ComponentChan <- containerComponent
 				// create the relation between the container and pod
 				cc.RelationChan <- cc.podToContainerStackStateRelation(pod.ExternalID, containerComponent.ExternalID)
+				// create the relation between the container and node
+				cc.RelationChan <- cc.containerToNodeStackStateRelation(containerComponent.ExternalID, nodeCorrelation.NodeExternalID)
```

Comment:
shouldn't it be `nodeCorrelation.NodeIdentifier` to match with the `containerComponent`'s identifer for sure?

Reply:
ah I see, `nodeCorrelation.NodeIdentifier` is used above to create containerComponent.ExternaID